telemetry/tracks/class-tracks-event.php

```diff
@@ -0,0 +1,336 @@
+		if ( is_wp_error( $this->data ) ) {
+			return $this->data;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Processes the event's properties to get them ready for validation.
+	 *
+	 * @param string $event_prefix The event's prefix.
+	 * @param string $event_name The event's name.
+	 * @param array<string, mixed>|array<empty> $event_properties Any event properties to be processed.
+	 * @return Tracks_Event_DTO The resulting event object with processed properties.
+	 */
+	protected static function process_properties(
+		string $event_prefix,
+		string $event_name,
+		array $event_properties
+	): Tracks_Event_DTO {
+		$event = static::encode_properties( $event_properties );
+		$event = static::set_user_properties( $event );
+
+		// Set event name. If the event name doesn't have the prefix, add it.
+		$event->_en = preg_replace(
+			'/^(?:' . $event_prefix . ')?(.+)/',
+			$event_prefix . '\1',
+			$event_name
+		) ?? '';
+
+		// Set event timestamp.
+		if ( ! isset( $event->_ts ) ) {
+			$event->_ts = static::milliseconds_since_epoch();
+		}
+
+		// Remove non-routable IPs to prevent record from being discarded.
+		if ( isset( $event->_via_ip ) &&
+			1 === preg_match( '/^192\.168|^10\./', $event->_via_ip ) ) {
+			unset( $event->_via_ip );
+		}
+
+		// Set VIP environment if it exists.
+		if ( defined( 'VIP_GO_APP_ENVIRONMENT' ) ) {
+			$app_environment = constant( 'VIP_GO_APP_ENVIRONMENT' );
+			if ( is_string( $app_environment ) && '' !== $app_environment ) {
+				$event->vipgo_env = $app_environment;
+			}
+		}
+
+		// Set VIP organization if it exists.
+		if ( defined( 'VIP_ORG_ID' ) ) {
+			$org_id = constant( 'VIP_ORG_ID' );
+			if ( is_string( $org_id ) && '' !== $org_id ) {
+				$event->vipgo_org = $org_id;
+			}
+		}
+
+		// Check if the user is a VIP user.
+		$event->is_vip_user = Support_User::user_has_vip_support_role( get_current_user_id() );
+
+		return $event;
+	}
+
+	/**
+	 * Sets the Tracks User ID and User ID Type depending on the current
+	 * environment.
+	 *
+	 * @param Tracks_Event_DTO $event The event to annotate with identity information.
+	 * @return Tracks_Event_DTO The new event object including identity information.
+	 */
+	protected static function set_user_properties( Tracks_Event_DTO $event ): Tracks_Event_DTO {
+		$wp_user = wp_get_current_user();
+
+		// Only track logged-in users.
+		if ( 0 === $wp_user->ID ) {
+			return $event;
+		}
+
+		// Set anonymized event user ID; it should be consistent across environments.
+		if ( defined( 'VIP_TELEMETRY_SALT' ) ) {
+			$salt           = constant( 'VIP_TELEMETRY_SALT' ); // this would be private but globally shared across the platform
+			$tracks_user_id = hash_hmac( 'sha256', $wp_user->user_email, $salt );
+
+			$event->_ui = $tracks_user_id;
+			$event->_ut = 'vip:user_email';
+
+			return $event;
+		}
+
+		// Users in the VIP environment.
+		if ( defined( 'VIP_GO_APP_ID' ) ) {
+			$app_id = constant( 'VIP_GO_APP_ID' );
+			if ( is_integer( $app_id ) && 0 < $app_id ) {
```

Comment:
Super minor nit, I find yoda-style checks for `<`/`>` hard to parse, and I need to transpose them in my head to read code. Could we:

```suggestion
			if ( is_integer( $app_id ) && $app_id > 0 ) {
```

Reply:
Sure, I find it hard to parse myself.